### ISSUE TITLE
xsv gets terminated with "Too many open file" error when doing partitioning

### ISSUE THREAD

**Issue by asahaf** [BurntSushi/xsv#196]
Hi,

I'm trying to partition a very big csv file (5119338774 rows), based on the first column. xsv gets terminated with Too many open files (os error 24) error.

Thanks,,

**Comment by BurntSushi** [comment]
I'd suggest increasing your open file descriptor limit for now.

**Comment by asahaf** [comment]
yeah, increasing the limit solves the issue. Thanks

should the partition sub-command have a flag or option for limiting the number of opened files? This is impotent especially when partitioning based on a sorted column, there is no need to keep files opened since we're not going to encounter those values again.

Thanks,

**Comment by BurntSushi** [comment]
Yes something should be done. But it's unlikely to happen soon. PRs are welcome.

**Comment by gyscos** [comment]
I started a fork to add this feature:
https://github.com/gyscos/xsv/commit/55028302a009ecc1fdffd5b59c40d481367de50f

It's still very much a WIP:
* It adds some `re_open` functions to open a file in append mode without truncating it. Not sure if separate functions like that are better than adding more options to the existing `writer` methods.
* It adds a dependency to the `lru` crate, which adds to build time.
* The `lru` crate also does not currently offer an Entry API, so we end up computing the hash 3 times.
* My editor runs `rustfmt` on save and it seems to have affected unrelated lines - I suspect the current files are not rustfmt'ed. A (c)leaner diff could be made by not reformatting the code.

I'm open to ways to improve this in case it could reach mergeable quality, or it could at least serve as inspiration for the next one to tackle this issue.
